### ISSUE TITLE
TVM sometimes running fast, sometimes running slow.

### ISSUE THREAD

**Issue by Cecilwang** [dmlc/tvm#997]
I wrote a simple demo and deployed native binary on android device. However, I got confused when I run my demo repeatedly from adb shell. The time spent ranged from 0.7ms to 4ms.

**Comment by tqchen** [comment]
The best way is to run it repetitively and take the mean. A proper way to do timing
- skip the first run and measurement because usually this is when opencl jit compilation happens
- do a tvm.cl(0).sync() to synchronize the device
- run several times after that and measure the remaining time cost.

This logic is implemented in tvm's time_evaluator and you can see if that produces more stable timing.

**Comment by dayanandasiet** [comment]
@Cecilwang 

Will you please share environment and target model details where benchmark is evaluating?

**Comment by Cecilwang** [comment]
Thanks for your suggestion @tqchen .  The problem has been solved.

@dayanandasiet, I deployed on HUAWEI mt7-tl10(CPU:Kirin 925 ) without OpenCL.
The model was created by Keras:

`from keras import backend as K
from keras.layers import Activation
from keras.layers import BatchNormalization
from keras.layers import Conv2D
from keras.layers import Input
from keras.models import Model
import numpy as np


def create_model():
    input_data = np.random.uniform(size=(1, 41, 23, 1))
    inputs = Input(tensor=K.constant(input_data))
    x = inputs
    x = Conv2D(32, (8, 8), padding="same", use_bias=True, strides=(2, 2))(x)
    x = BatchNormalization(axis=-1)(x)
    x = Activation(K.relu)(x)
    model = Model(inputs=inputs, outputs=x)`
``
